### ISSUE TITLE
DefinitionResponse selects full type range

### ISSUE THREAD

**Issue by jrieken** [Microsoft/TypeScript#8808]
When sending a `definition` request to the tsserver, the response contains a range selection the whole type. For UX proposes it would be better to either select only the identifying portion, like a type name, or additional include such an identifying range.

See https://github.com/Microsoft/vscode/issues/6772 background.

**Comment by jrieken** [comment]
@mhegazy ping

**Comment by mhegazy** [comment]
I do not see why not. @CyrusNajmabadi any objections to changing `DefinitionInfo.textSpan` to be the declaration name span instead of the whole declaration span?

**Comment by CyrusNajmabadi** [comment]
There shouldn't be any problem doing this.  That said, i'm somewhat surprised that things work this way.  When we goto-def on a type in TS today, we don't go to the identifier?

**Comment by mhegazy** [comment]
> we don't go to the identifier?

nope. the beginning of the declaration. usually that is not too far from the identifier anyways.
